### PR TITLE
add the ability to configure app-insights export

### PR DESCRIPTION
**Pull request by bmc-msft**
## Summary of the Pull Request

Adds the ability to enable automatic log export from the instance specific appinsights

## PR Checklist
* [x] Applies to work item: #44 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

Adds the `--export_appinsights` to the deployment script to setup app insight exports to the `app-insights` container on the `func` storage account.  Note, it will automatically replace any existing app-insights exports to the same container prior to creating the export config.

## Validation Steps Performed

Deploy with `--export_appinsights`, run for a while.  Note new files are created in `app-insights` container on `func` storage account.

### CODE REVIEW

File: src/deployment/deploy.py [hunk 1]
```diff
@@ -383,6 +389,68 @@ def create_eventgrid(self):
...
+
+        logger.info("adding appinsight log export")
+        account_name = self.results["deploy"]["func-name"]["value"]
+        key = self.results["deploy"]["func-key"]["value"]
+        account_url = "https://%s.blob.core.windows.net" % account_name
+        client = BlobServiceClient(account_url, credential=key)
+        if container_name not in [x["name"] for x in client.list_containers()]:
+            client.create_container(container_name)
+
+        expiry = datetime.utcnow() + timedelta(days=2 * 365)
```
Comment (ranweiler) [inline src/deployment/deploy.py:407]: Can we accomplish this with something other than an ad hoc (thus irrevocable) SAS URL?

Comment (ranweiler) [reply src/deployment/deploy.py:407]: Alternatively: use a much shorter-lived URL, and have a mechanism for refreshing within an instance (though that feels a bit roundabout).

Comment (bmc-msft) [reply src/deployment/deploy.py:407]: The API unfortunately requires SAS urls.

Doing it through the instance requires granting more privileges to the service.

Comment (ranweiler) [reply src/deployment/deploy.py:407]: Okay. Given the restrictions on perms, and the fact that none of our code touches it (it is only given directly to an Azure API, here), this seems okay to me. Can we just document our usage assumptions? In particular, note that this SAS URL must not "leave this script", except to the extent that it implicitly does in its use in the export request.

**Review by ranweiler — APPROVED** [review]

**Pull request merged**